### ISSUE TITLE
Vagrant install isn't playing ball

### ISSUE THREAD

**Issue by x-light** [churchio/onebody#493]
**TL;DR**
I can't get any rake tasks to run after install on vagrant. Error looks like:
```
vagrant@ubuntu-14:/vagrant/config$ rake secret
/home/vagrant/.rvm/rubies/ruby-2.1.5/lib/ruby/site_ruby/2.1.0/rubygems/specification.rb:2716:in `stat': No such file or directory @ rb_file_s_stat - README (Errno::ENOENT)
```
----
Today I've attempted a clean install of OneBody for development purposes using vagrant. I'm using Ubuntu 14.04. Here's a summary of pertinent stuff I noticed (which might be completely useless! ;)

* The same error as noted above also affects the rails server so I'm pretty sure its not starting either.

* When I go to localhost:8080/, I get the apache web server page. Should I expect this?

* Checked the database thru mysql and it doesn't seem to have any tables created (eg. run select * from groups returns "table not found: onebody_dev.groups". (I'm not sure if I'm meant to run the rake tasks to create the database or whether that gets run in the install)

* I got a rubygems failed to compile error when the script called bundle install:
```
==> default: Gem::RemoteFetcher::UnknownHostError: no such name (https://rubygems.org/gems/currencies-0.4.2.gem)
==> default: An error occurred while installing currencies (0.4.2), and Bundler cannot
==> default: continue.
==> default: Make sure that `gem install currencies -v '0.4.2'` succeeds before bundling.
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

```
so I just logged into the vagrant box and ran bundle install, and all was rosy.

Heres a [gist of my install process](https://gist.github.com/x-light/5ebf0d27d1de91fc24c0) in case its useful. There will be a pop quiz afterwards ;).

cc @attieretief

**Comment by x-light** [comment]
I think this might be a bundler problem as I've just experienced this same thing with my local install. which picked up bundler 1.10.1. 

I downgraded to bundler 1.9.4 and the problem seemed to "go away" ...

Hold this for now and I'll update further when i retry the vagrant machine.

**Comment by attieretief** [comment]
@x-light I also tried to start from scratch...
This is a far as I got - I confirm a bundler issue
See last section of my gist...
https://gist.github.com/attieretief/0467506be575e415c433

**Comment by x-light** [comment]
aha - and I see bundler has released a minor update inbetween me and you running the installs. (Your version is 1.10.2 & mine was 1.10.1)

I'll have a bit more of a go with this today and see if I can find a resolution. Might just be a case of downgrading bundler for now.

**Comment by x-light** [comment]
@attieretief something strange has happened with my install of vagrant. I wound up installing onebody locally in the end, as I already have ruby, git and the like on my machine.

My suspicion is that you should be able to:

```
gem uninstall bundler 
```
which will get rid of 1.10.2. You'll get dependency errors, just be brave and say OK. Then: 

```
gem install bunder -v 1.9.4
```

and all should be rosy to carry on after that. You'll be at step 10 of the manual install process so you'll just need to run the last few commands on the vagrant box.

Let us know how you go! :)

**Comment by attieretief** [comment]
Thanks @x-light. I won't have much time to tinker with it this week, I'll just use the virtual appliance for now. Will let you know once I try again.

**Comment by edelgado** [comment]
Thanks. I modified `Vagrantfile` like this and made everything work for my clean install:
```
diff --git a/Vagrantfile b/Vagrantfile
index a9d8877..541c1d5 100644
--- a/Vagrantfile
+++ b/Vagrantfile
@@ -46,7 +46,7 @@ user=$(cat <<USER

   # bundle gems
   cd /vagrant
-  gem install bundler --no-ri --no-rdoc
+  gem install bundler -v 1.9.4 --no-ri --no-rdoc
   if [[ ! -e config/database.yml ]]; then
     cp config/database.yml{.mysql-example,}
   fi
@@ -94,7 +94,7 @@ SCRIPT
```

**Comment by seven1m** [comment]
@edelgado would you make a PR with that change and I'll merge. Thanks!

**Comment by x-light** [comment]
thanks @edelgado!

**Comment by edelgado** [comment]
Sure thing, PR coming up!

**Comment by edelgado** [comment]
Here ya go: https://github.com/churchio/onebody/pull/499

**Comment by attieretief** [comment]
Nice work guys. I tried vagrant again this morning - but now keep running into the following issue:

Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'phusion/ubuntu-14.04-amd64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'phusion/ubuntu-14.04-amd64' is up to date...
==> default: Setting the name of the VM: onebody_default_1433307691176_58707
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 80 => 8080 (adapter 1)
    default: 22 => 2222 (adapter 1)
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
    default: 
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default: 
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if its present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
==> default: Mounting shared folders...
    default: /vagrant => /Users/aretief/onebody
==> default: Running provisioner: shell...
    default: Running: inline script
==> default: stdin: is not a tty
==> default: ++ apt-get update -qq
==> default: W: Failed to fetch gzip:/var/lib/apt/lists/partial/us.archive.ubuntu.com_ubuntu_dists_trusty-updates_universe_i18n_Translation-en  Hash Sum mismatch
==> default: 
==> default: E: Some index files failed to download. They have been ignored, or old ones used instead.
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

**Comment by x-light** [comment]
Hi @attieretief, I asked Aunty Google on this one as it looks like an Ubuntu error.
Have a look here: http://askubuntu.com/questions/41605/trouble-downloading-packages-list-due-to-a-hash-sum-mismatch-error 
Maybe try the remove apt-partials directory option? If no good, then we might be able to point specifically to a SAF based server.

**Comment by attieretief** [comment]
thanks @x-light - I just added this before the apt-get update -qq line in the vagrant file:

sudo sed -i 's/us.archive.ubuntu.com/za.archive.ubuntu.com/' /etc/apt/sources.list

to point it to the za archive, and vagrant up now completes

**Comment by x-light** [comment]
Great @attieretief. I'll close this now as we've overcome bundler, apt-get and vagrant's attempts to thwart your onebody install :smile:
